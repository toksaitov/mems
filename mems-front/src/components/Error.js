import React from 'react';

const Error = ({ error, clearError }) => error ?
    <div className="row justify-content-center mt-3">
        <div className="col-md-5">
            <div className="alert alert-danger">
                {error}
                <button type="button" className="close" aria-label="Close" onClick={clearError}>
                    <span aria-hidden="true">&times;</span>
                </button>
            </div>
        </div>
    </div>
    :
    <></>;

export default Error;
