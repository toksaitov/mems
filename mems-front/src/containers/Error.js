import { connect } from 'react-redux';

import { clearError } from '../actions/error.js';

import Error from '../components/Error.js';

const mapStateToProps = state => ({
    'error': state.error
});

const mapDispatchToProps = dispatch => ({
    'clearError': () => dispatch(clearError())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Error);
