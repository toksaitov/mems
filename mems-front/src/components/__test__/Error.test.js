import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Error from '../Error.js';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Error />, div);
});

it('shows a specific error message', () => {
    const wrapper = shallow(<Error error="Something went wrong" />);
    expect(wrapper.contains("Something went wrong")).toBe(true);
});

it('matches the previous snapshot', () => {
    const tree = renderer.create(<Error error="test message" />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('clears error', () => {
    let cleared = false;
    const clearErrorFake = () => { cleared = true; };

    const wrapper = shallow(<Error error="test message" clearError={clearErrorFake} />);
    wrapper.find('button').simulate('click');

    expect(cleared).toBe(true);
});

