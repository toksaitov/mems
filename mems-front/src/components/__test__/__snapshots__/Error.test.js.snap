// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot 1`] = `
<div
  className="row justify-content-center mt-3"
>
  <div
    className="col-md-5"
  >
    <div
      className="alert alert-danger"
    >
      test message
      <button
        aria-label="Close"
        className="close"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          Ã—
        </span>
      </button>
    </div>
  </div>
</div>
`;
